### PR TITLE
enable async for update tsvs [AS-916]

### PR DESCRIPTION
**Pull request by davidangb**
Enables async imports for update TSVs, where previously we only supported entity and membership TSVs.

### CODE REVIEW

File: src/main/scala/org/broadinstitute/dsde/firecloud/dataaccess/HttpImportServiceDAO.scala [hunk 1]
```diff
@@ -21,18 +21,18 @@ class HttpImportServiceDAO(implicit val system: ActorSystem, implicit val materi
   extends ImportServiceDAO with RestJsonClient with SprayJsonSupport {
 
   override def importPFB(workspaceNamespace: String, workspaceName: String, pfbRequest: AsyncImportRequest)(implicit userInfo: UserInfo): Future[PerRequestMessage] = {
-    doImport(workspaceNamespace, workspaceName, pfbRequest, "pfb")
+    doImport(workspaceNamespace, workspaceName, isUpsert = true, pfbRequest, "pfb")
```
Comment (davidangb) [inline src/main/scala/org/broadinstitute/dsde/firecloud/dataaccess/HttpImportServiceDAO.scala:24]: PFB imports are always upsert.

File: src/main/scala/org/broadinstitute/dsde/firecloud/EntityService.scala [hunk 2]
```diff
@@ -201,23 +201,17 @@ class EntityService(rawlsDAO: RawlsDAO, importServiceDAO: ImportServiceDAO, goog
...
-      if (!isUpsert) {
-        // until Rawls & Import Service implement update-only TSVs, throw a BadRequest if the user requested async
-        // but sent an update TSV.
-        Future(RequestCompleteWithErrorReport(BadRequest, "Update-only TSVs cannot use async mode"))
-      } else {
-        asyncUpsert(workspaceNamespace, workspaceName, rawlsCalls, userInfo).recover {
-          case e: Exception =>
-            RequestCompleteWithErrorReport(InternalServerError, "Unexpected error during async TSV import", e)
-        }
+      asyncUpsert(workspaceNamespace, workspaceName, isUpsert, rawlsCalls, userInfo).recover {
```
Comment (ansingh7115) [inline src/main/scala/org/broadinstitute/dsde/firecloud/EntityService.scala]: Should we change code comments on lines 201-202?

**Review by MatthewBemis — APPROVED** [review]

File: src/main/scala/org/broadinstitute/dsde/firecloud/EntityService.scala (hunk 2 above)
Comment (davidangb) [reply src/main/scala/org/broadinstitute/dsde/firecloud/EntityService.scala]: yes, we should! Good catch; removed

File: src/main/scala/org/broadinstitute/dsde/firecloud/EntityService.scala [hunk 3]
```diff
@@ -201,23 +201,17 @@ class EntityService(rawlsDAO: RawlsDAO, importServiceDAO: ImportServiceDAO, goog
...
+          RequestCompleteWithErrorReport(InternalServerError, "Unexpected error during async TSV import", e)
       }
     } else {
       val rawlsResponse = rawlsDAO.batchUpsertEntities(workspaceNamespace, workspaceName, entityType, rawlsCalls)(userInfo)
       handleBatchRawlsResponse(entityType, rawlsResponse)
     }
   }
 
-  private def asyncUpsert(workspaceNamespace: String, workspaceName: String,
+  private def asyncUpsert(workspaceNamespace: String, workspaceName: String, isUpsert: Boolean,
```
Comment (sagehen03) [inline src/main/scala/org/broadinstitute/dsde/firecloud/EntityService.scala]: Seeing a method called `asyncUpsert` with a parameter called `isUpsert` is jarring to me.  It's called `asyncUpsert` so  `isUpsert` should always be true at least for the easiest comprehension.  I understand that the isUpsert name is baked in to Rawls and ImportService already so doing something like changing the param to `isUpdateOnly` and inverting the logic here in Orch will make the code less consistent with Rawls and ImportService.  What if we renamed this method to `asyncUpdateOrInsert` or just just `asyncUpdate`?  I'll leave it to you to decide if there's a sensible way to address my overloaded upsert hangup.

**Review by sagehen03 — APPROVED** [review]
One question around our overloaded use of "upsert", but not a blocker.

**Review by ansingh7115 — APPROVED** [review]

File: src/main/scala/org/broadinstitute/dsde/firecloud/EntityService.scala (hunk 3 above)
Comment (davidangb) [reply src/main/scala/org/broadinstitute/dsde/firecloud/EntityService.scala]: no, that's good feedback, and my fault - I had that method name left over from when it truly was upsert-only, and failed to rename it!

Also, wow, your comment called out another problem in the synchronous path, which I am fixing too.

Comment (sagehen03) [reply src/main/scala/org/broadinstitute/dsde/firecloud/EntityService.scala]: Cool, good catch. High five! :hand:

**Review by sagehen03 — APPROVED** [review]

**Pull request merged**